ONSdigital/eq-terraform
# Eq 18 add elastic beanstalk support

**What?**

Within the context of story EQ-18, this story's aim was to automate the creation of EQ infrastructure and to deploy the key components of the system currently; namely the survey runner skeleton, the rabbitmq cluster and the submitter daemon system, hosting all of the components in a secure VPC within AWS.

This PR refactors the terraform deployment to create a VPC into which we create a new subnet, internet gateway and launch our rabbitmq servers and eq-submitter servers into. This PR explicitly allows internal VPC traffic to the rabbitmq servers including clustering and amqp connections. We may choose to lock this down further in future PR's.

We currently still allow ssh connections to the rabbitmq instances for provisioning but we will need to remove this later, moving the provisioning to work via a bastion box or explicit IP range.

**How to test**

_Before you begin_
- Install golang, git & mercurial via brew / apt-get
-  git clone terraform from github.com/hashicorp/terraform into `$GOPATH/src/github.com/hashicorp/terraform.`
- cd into $GOPATH/src/github.com/hashicorp/terraform
- Checkout the elasticbeanstalk feature branch `f-aws-elastic-beanstalk`
- Run `make updatedeps`
- Run `make dev`
- Use the created terraform binary (`bin/terraform') for testing

_Testing_
1. Run `bin/terraform` apply --var "env=test"
2. Wait till the deployment has finished and then visit the outputted r53 url for the survey runner.
3. Check that the healthcheck (at /healthcheck) returns 200 ok HTTP status code
4. Have a look at the created resources in the AWS console / control panel, doing some exploratory testing of environment variables being set and the correct urls / names being used for various components.

**Outstanding on EQ-18**
- Automate submitter deployment, see separate PR.

**Who can test**
Any one but @dhilton
